network_test.go

```diff
@@ -185,3 +185,50 @@ func TestUpdateRoutes(t *testing.T) {
 	assert.True(t, reflect.DeepEqual(results.Routes[0], testRoutes.Routes[1]),
 		"Interface created didn't match: got %+v, expecting %+v", results.Routes[0], testRoutes.Routes[1])
 }
+
+func TestListInterfaces(t *testing.T) {
+	skipUnlessRoot(t)
+
+	s := sandbox{}
+
+	ifc := pb.Interface{
+		Name:   "enoNumber",
+		Mtu:    1500,
+		HwAddr: "02:00:ca:fe:00:48",
+	}
+	ip := pb.IPAddress{
+		Family:  0,
+		Address: "192.168.0.101",
+		Mask:    "24",
+	}
+	ifc.IPAddresses = append(ifc.IPAddresses, &ip)
+
+	netHandle, _ := netlink.NewHandle()
+	defer netHandle.Delete()
+
+	// create a dummy link that we can test update on
+	macAddr := net.HardwareAddr{0x02, 0x00, 0xCA, 0xFE, 0x00, 0x48}
+	link := &netlink.Dummy{
+		LinkAttrs: netlink.LinkAttrs{
+			MTU:          1500,
+			TxQLen:       -1,
+			Name:         "ifc-name",
+			HardwareAddr: macAddr,
+		},
+	}
+	netHandle.LinkAdd(link)
+	netHandle.LinkSetUp(link)
+
+	resultingIfc, err := s.updateInterface(netHandle, &ifc)
```

Comment:
This whole test might run on a dev computer, and I am wondering if this is safe to do such a thing?
I mean we're basically updating the network and then checking. We should run this test into its own network namespace to check the function easily.
Basically, we enter the netns (need to be safe about it, meaning no go routine as they will jump to a different netns), then we create a few links, and then we run `listInterfaces()` to check the list of interfaces we actually found.